gradle/gradlew-include.sh

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (C) 2024 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Downloads the gradle-wrapper.jar if necessary and verifies its integrity.
+# Included from /.gradlew
+
+# Extract the Gradle version from gradle-wrapper.properties.
+GRADLE_DIST_VERSION="$(grep distributionUrl= "$APP_HOME/gradle/wrapper/gradle-wrapper.properties" | sed 's/^.*gradle-\([0-9.]*\)-[a-z]*.zip$/\1/')"
+GRADLE_WRAPPER_SHA256="$APP_HOME/gradle/wrapper/gradle-wrapper-${GRADLE_DIST_VERSION}.jar.sha256"
+GRADLE_WRAPPER_JAR="$APP_HOME/gradle/wrapper/gradle-wrapper.jar"
+if [ ! -e "${GRADLE_WRAPPER_SHA256}" ]; then
+  # Delete the wrapper jar, if the checksum file does not exist.
+  rm -f "${GRADLE_WRAPPER_JAR}"
+fi
+if [ -e "${GRADLE_WRAPPER_JAR}" ]; then
+  # Verify the wrapper jar, if it exists, delete wrapper jar and checksum file, if the checksums
+  # do not match.
+  JAR_CHECKSUM="$(sha256sum "${GRADLE_WRAPPER_JAR}" | cut -d\  -f1)"
+  EXPECTED="$(cat "${GRADLE_WRAPPER_SHA256}")"
+  if [ "${JAR_CHECKSUM}" != "${EXPECTED}" ]; then
+    rm -f "${GRADLE_WRAPPER_JAR}" "${GRADLE_WRAPPER_SHA256}"
+  fi
+fi
+if [ ! -e "${GRADLE_WRAPPER_SHA256}" ]; then
+  curl --location --output "${GRADLE_WRAPPER_SHA256}" https://services.gradle.org/distributions/gradle-${GRADLE_DIST_VERSION}-wrapper.jar.sha256 || exit 1
+fi
+if [ ! -e "${GRADLE_WRAPPER_JAR}" ]; then
+  # The Gradle version extracted from the `distributionUrl` property does not contain ".0" patch
+  # versions. Need to append a ".0" in that case to download the wrapper jar.
+  GRADLE_VERSION="$(echo "$GRADLE_DIST_VERSION" | sed 's/^\([0-9]*[.][0-9]*\)$/\1.0/')"
+  curl --location --output "${GRADLE_WRAPPER_JAR}" https://raw.githubusercontent.com/gradle/gradle/v${GRADLE_VERSION}/gradle/wrapper/gradle-wrapper.jar || exit 1
```

Comment:
justFYI:

the gradle wrapper jar under that URL is never the official one from the release:

see git history:
https://github.com/gradle/gradle/commits/v8.9.0/gradle/wrapper

so if we are unlucky and the jar changes in a new release, the sha256 check will probably be failing.
as of right now, we are in luck though:
```
==> gradle-8.9-rc-1-wrapper.jar.sha256 <==
498495120a03b9a6ab5d155f5de3c8f0d986a449153702fb80fc80e134484f17
==> gradle-8.9-rc-2-wrapper.jar.sha256 <==
498495120a03b9a6ab5d155f5de3c8f0d986a449153702fb80fc80e134484f17
==> gradle-8.9-wrapper.jar.sha256 <==
498495120a03b9a6ab5d155f5de3c8f0d986a449153702fb80fc80e134484f17%             
```

Reply:
That behavior is weird, but correct. The jar in there is the one that makes it into the Gradle release (`gradle wrapper` "builds" it from that one).
I have no clue why Gradle doesn't publish `gradle-wrapper.jar`s though. It's disappointing.

Reply:
BTW: That approach is used in many Apache projects - nobody complained so far AFAIK

Reply:
afaik the wrapper is build from scratch:
https://github.com/gradle/gradle/blob/master/platforms/core-runtime/wrapper-main/build.gradle.kts#L69-L85
the commited `gradle/wrapper/gradle-wrapper.jar` is what bootstraps the build of gradle though (kind of a hen egg problem)

but yeah i think there is a ticket with many interested parties asking for an official download link of the wrapper jar...

Reply:
It's _re_-built, yes - but deterministically